### DISCUSSION TITLE
No recibe correctamente los INPUT

### DISCUSSION THREAD

**Discussion by Osiries1111** [IIC2133-PUC/2024-2#29] [Tarea 0] [answered]
Hola! 
Tengo el problema que no se me reciben correctamente algunos INPUT en algunas funciones, como el BUY y el STEAL (no se si el GIVE me funciona bien porque el código me arroja SEGMENTATION FAULT antes de llegar a este punto) , cuando imprimo lo que me recibe, me coloca algo muy distinto a lo que se espera, como puede ser, que me da un 0 pero debió ser un 2 , llegando a ser números arbitrarios como 25959 cuando debería ser un 12.

Espero que me puedan dar alguna solución para esto...

![image](https://github.com/user-attachments/assets/05ffeac2-b680-4304-935e-2946e46132c0)
![image](https://github.com/user-attachments/assets/e51de43d-cee2-4430-8efb-05398a095b15)

**Comment by Osiries1111** [comment, marked as answer]
Hola!
Lo pude arreglar por mi cuenta. La solución y raíz del problema fue el siguiente:

Para transformar los colores a números estaba utilizando (int)color, el cual me convierte el color a su código ASCII, el problema estaba en el stack o algo así por el estilo (me corrigen si estoy mal, no soy experta en el asunto) por lo que algunos números como que no se guardaban bien el stack de la memoria, por los tamaños variantes de estos.
Por ello, realicé variables char color[10] ,para recibirlos de tamaño a lo más 10 cada color, y los mapeé a una lista que me cree para los colores, y así tener la "traducción" de cada color a número.

Todavía no entiendo del todo porqué esto me solucionó el problema, pero fue eso básicamente lo que lo arregló.
Si alguien sabe más acerca del tema, estaría bacán :)

**Comment by CataAlamos** [comment]
Hola, tu error se arregló ya que color te lo están pasando como char y no int, entonces si dices que es un int el programa lo transforma a int. Puedes guardarlo como int si haces un tipo de diccionario, tipo: 
`enum Colors {
    YELLOW = 1, 
    RED = 2, 
    BLUE = 3,
    GREEN = 4
}; `
Espero haber respondido bien tu pregunta
